src/main/scala/com/amazon/deequ/analyzers/Distance.scala

```diff
@@ -40,32 +52,198 @@ object Distance {
+   * @param percThresholdYates           Yates percentage of categories that can be below threshold for tables larger than 2x2
+   * @param absThresholdCochran          Cochran absolute threshold for 2x2 tables
+   * @return distance                    can be an absolute distance or a p-value based on the correctForLowNumberOfSamples argument
+   */
+  def categoricalDistance(
+    sample1: scala.collection.mutable.Map[String, Long],
+    sample2: scala.collection.mutable.Map[String, Long],
+    correctForLowNumberOfSamples: Boolean = false,
+    method: CategoricalDistanceMethod = LInfinity,
+    absThresholdYates: Integer = 5,
+    percThresholdYates: Double = 0.2,
+    absThresholdCochran: Integer = 10)
     : Double = {
+    method match {
+      case LInfinity => categoricalLInfinityDistance(sample1, sample2, correctForLowNumberOfSamples)
+      case Chisquare => categoricalChiSquareTest(sample1, sample2, correctForLowNumberOfSamples, absThresholdYates , percThresholdYates,absThresholdCochran )
+    }
+  }
+
+  /** Calculate distance of categorical profiles based on Chisquare test or stats
+   *
+   *  for 2x2 tables: all expected counts should be 10 or greater (Cochran, William G. "The χ2 test of goodness of fit." The Annals of mathematical statistics (1952): 315-345.)
+   *  for tables larger than 2 x 2: "No more than 20% of the expected counts are less than 5 and all individual expected counts are 1 or greater" (Yates, Moore & McCabe, 1999, The Practice of Statistics, p. 734)
+   *
+   *  @param sample                       the mapping between categories(keys) and counts(values) of the observed sample
+   *  @param expected                     the mapping between categories(keys) and counts(values) of the expected baseline
+   *  @param correctForLowNumberOfSamples if true returns chi-square statistics otherwise p-value
+   *  @param absThresholdYates            Yates absolute threshold for tables larger than 2x2
+   *  @param percThresholdYates           Yates percentage of categories that can be below threshold for tables larger than 2x2
+   *  @param absThresholdCochran          Cochran absolute threshold for 2x2 tables
+   *  @return distance                    can be an absolute distance or a p-value based on the correctForLowNumberOfSamples argument
+   *
+   */
+  private[this] def categoricalChiSquareTest(
+    sample: scala.collection.mutable.Map[String, Long],
+    expected: scala.collection.mutable.Map[String, Long],
+    correctForLowNumberOfSamples: Boolean = false,
+    absThresholdYates : Integer = 5 ,
+    percThresholdYates : Double = 0.2,
+    absThresholdCochran : Integer = 10)
+  : Double = {
 
-      var n = 0.0
-      var m = 0.0
-      sample1.keySet.foreach { key =>
-        n += sample1(key)
+    val sample_sum: Double = sample.filter(e => expected.contains(e._1)).map((e => e._2)).sum
+    val expected_sum: Double = expected.map(e => e._2).sum
```

Comment:
Can we change these to camel case? There are other occurrences of snake case in the rest of the file. Please change to camel case wherever applicable.

Reply:
changed to camel case